### DISCUSSION TITLE
Toggling visibility of text in editor for custom file extension

### DISCUSSION THREAD

**Discussion by NotoriousBBQ** [microsoft/vscode-discussions#1066] [Extension Development QnA] [answered]
We have a custom script file format (essentially a glorified text format) for which we have already developed a syntax highlighting extension.  A language server is also in development.

I would like to display some portion of a script line in one editor view, and have it be invisible in others.  I'm not adverse to having separate view panel for this, or using a command toggle it.  

What would be the best way to accomplish that?  I'm new to VS Code extensions, so I may have missed something simple. :)

A bit more background.  The script contains display strings that get placed into a Unity string table in an external process. That requires a unique identifier (string) associated with each line of display text.   I'd like to hide that identifier in the script writer view so it isn't cluttered, but show it in developer view for tracking purposes.

Thanks!

**Comment by alefragnani** [comment, marked as answer]
Hi @NotoriousBBQ ,

Are you looking for a way to _hide words/sentences_ similar to those "Top Secret" files that hide sensitive information with black stripes, or _fold_ a group of lines?

If you are looking for hiding words/sentences, you may look the different approaches used by these two extensions:
* https://marketplace.visualstudio.com/items?itemName=dotenv.dotenv-vscode
* https://marketplace.visualstudio.com/items?itemName=johnpapa.vscode-cloak

The first one uses the Decorator API (https://github.com/microsoft/vscode-extension-samples/tree/main/decorator-sample) and the second one is based on the TextMate grammar elements.

On the other hand, if you are looking for the folding approach, you could take a look at the Folding API https://code.visualstudio.com/api/language-extensions/language-configuration-guide#folding.

Hope this helps

**Reply by NotoriousBBQ** [reply]
Thank you!  I'm looking for the 'Top Secret' approach.  I had considered folding, but the hidden elements are not arranged in a way that would make folding efficient.  Though editor.foldingStrategy _might_ be flexible enough once our language server is stable. 

Of the 2 you 'Top Secret' extensions you suggest, cloak seems like the better fit.  I'm not worried about the _content_ of the hidden elements, I just want them invisible for users that don't care.  

I'll give cloak a look see.

Thank you again.
